File: OJ/usaco/cpp/beads.cpp
/*
ID:oifox20071
LANG:C++
PROG:beads
*/
#include <iostream>
using namespace std;
const
	int maxn = 350;
int n;
char s[maxn];
int main()
{
	freopen("beads.in", "r", stdin); freopen("beads.out", "w", stdout);
	
	cin >> n;
	int i;
	for (i = 0; i < n; ++ i)
	{
		cin >> s[i];
		s[i + n] = s[i];
	}
	char c;
	int flag, total, j;
	int max = 0;
	for (i = 0; i < n; ++ i)
	{
		c = s[i];
		if (c == 'w') flag = 0; else flag = 1;
		total = 0;
		j = i;
		while (flag <= 2)
		{
			while ((j < n + i) && (s[j] == c || s[j] == 'w'))
			{
				++ j;
				++ total;
			}
			c = s[j];
			++ flag;
		}
		if (total > max) max = total;
	}
	cout << max << endl;
	return 0;
}

File: OJ/usaco/pascal/s1/s11/beads.pas
{
ID:lijianx1
PROG:beads
LANG:PASCAL
}
program beads;
const maxn=350;
var a:array[1..maxn*2] of char;
    i,n,k,max:longint;
function right(x:longint):longint;
 var i,total:longint;
     now:char;
 begin
      total:=0; now:='p';
      for i:=x to x+n-1 do
         if a[i]='w' then inc(total)
	      else if now='p' then begin
	         inc(total); now:=a[i];
              end else if now=a[i] then inc(total)
		      else exit(total);
 end;

function left(x:longint):longint;
 var i,k,total:longint;
     now:char;
 begin
     i:=x-1; now:='p'; k:=0; total:=0;
     while k<=n do begin
         if i=0 then i:=2*n; inc(k);
	 if a[i]='w' then inc(total)
             else if now='p' then begin
	        inc(total); now:=a[i];
	     end else if now=a[i] then inc(total)
	        else exit(total);
	 dec(i);
     end;
 end;

begin
    assign(input,'beads.in'); reset(input);
    readln(n);
    for i:=1 to n do begin
	read(a[i]);
	a[i+n]:=a[i];
    end;
    close(input);
    max:=0;
    for i:=1 to n do begin
        k:=left(i)+right(i);
	if k>=n then k:=n;
	if k>max then max:=k;
    end;
    assign(output,'beads.out'); rewrite(output);
    writeln(max);
    close(output);
end.
